packages/nx/src/tasks-runner/create-task-graph.ts

```diff
@@ -490,3 +434,57 @@ function interpolateOverrides<T = any>(
   });
   return interpolatedArgs;
 }
+
+/**
+ * This function is used to filter out the dummy tasks from the dependencies
+ * It will manipulate the dependencies object in place
+ */
+export function filterDummyTasks(dependencies: { [k: string]: string[] }) {
+  const cycles = new Set(findCycles({ dependencies }).flat());
```

Comment:
Rather then creating `string[][]`, flattening, then creating a set, could we return an unique list from `findCycles`?

Reply:
change the return type to be Set<string>